ci/build_setup.sh

```diff
@@ -24,6 +24,15 @@ function setup_clang_toolchain() {
   echo "$CC/$CXX toolchain configured"
 }
 
+function setup_libcpp() {
+  # Force cc_wrapper and recipe-based dependencies to link against libc++.
+  export CXXFLAGS="-stdlib=libc++"
+  # Force Bazel to link using lld, since GNU ld has trouble linking against libc++fs.
+  export BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS} --linkopt -fuse-ld=lld"
+  export BAZEL_TEST_OPTIONS="${BAZEL_TEST_OPTIONS} --linkopt -fuse-ld=lld"
+  echo "libc++ configured"
```

Comment:
Meta-question; do we have upstream Bazel tracking for a better way to select libc++ vs. libstdc++? Even if we can do what is needed in this PR, I think Bazel should add much cleaner support to handle this, it should be a simple CLI switch..

Reply:
I've added it to https://github.com/bazelbuild/bazel/issues/6926, since it seems that `cc_configure` is undergoing major rewrite.

Reply:
I'll also add this to our Bazel team biweekly sync, thanks.